src/services/dashboard/modules/ResourceMap.vue

```diff
@@ -103,142 +116,78 @@ export default {
             default: () => ({}),
         },
     },
-    setup(props) {
-        const queryHelper = new QueryHelper();
-        const apiQuery = new ApiQueryHelper();
 
+    setup() {
         const state = reactive({
             chartRef: null as HTMLElement | null,
             chartRegistry: {},
-            chart: null as null|any,
-            data: [] as any,
-            filteredData: [] as any,
-            selectedProvider: '', // aws
-            selectedRegion: '', // Asia Pacific (Seoul)
-            selectedRegionCode: '', // ap-northeast-2
-            providers: computed(() => store.state.reference.provider.items),
-            regionList: [],
+            chart: null as MapChart|null,
+            data: [] as Resource[],
             loading: true,
+            providers: computed(() => store.state.reference.provider.items),
+            regions: computed(() => store.state.reference.region.items),
+            resourceDataByRegion: [] as CloudService[],
+            selectedRegionLabel: '',
+            selectedRegionCode: '',
+            selectedProvider: '',
             maxValue: 0,
-            initialRegion: {} as any,
         });
 
         const chartState = reactive({
-            registry: {},
-            chartData: [] as object[],
-            skeletons: range(5),
-            providers: computed(() => state.data.map(d => ({
-                name: state.providers[d.provider]?.label,
-                color: state.providers[d.provider]?.color as string,
-            }))
-                .filter((data, index, arr) => arr.findIndex(item => item.name === data.name) === index)),
             marker: null,
+            initialRegion: computed(() => ({
+                title: state.regions[state.data[0]?.region_code]?.name || state.data[0]?.region_code,
+                longitude: state.data[0]?.longitude,
+                latitude: state.data[0]?.latitude,
+                color: state.providers[state.data[0]?.provider]?.color as string,
```

Comment:
```suggestion
                color: state.providers[state.data[0]?.provider]?.color ?? '',
```

타입 이슈생겨서 이렇게 하신 것 같아서..
타입 이슈를 무조건 as string으로 처리하는건 좋지않은것 같아 위 방식 제안드려봅니다..ㅎ

Reply:
오 알아갑니다!